support/builders/tow-boot/default.nix

```diff
@@ -64,201 +64,228 @@ let
+      ''}
+      ${lib.optionalString withTTF ''
+      # Truetype console configuration
+      CONFIG_CONSOLE_TRUETYPE=y
+      CONFIG_CONSOLE_TRUETYPE_NIMBUS=y
+      CONFIG_CONSOLE_TRUETYPE_SIZE=26
+      # Ensure the chosen font is used
+      CONFIG_CONSOLE_TRUETYPE_CANTORAONE=n
+      CONFIG_CONSOLE_TRUETYPE_ANKACODER=n
+      CONFIG_CONSOLE_TRUETYPE_RUFSCRIPT=n
+      ''}
+
+      ${lib.optionalString withLogo ''
+      # For the splash screen
+      CONFIG_CMD_BMP=y
+      CONFIG_SPLASHIMAGE_GUARD=y
+      CONFIG_SPLASH_SCREEN=y
+      CONFIG_SPLASH_SCREEN_ALIGN=y
+      CONFIG_VIDEO_BMP_GZIP=y
+      CONFIG_VIDEO_BMP_LOGO=y
+      CONFIG_VIDEO_BMP_RLE8=n
+      CONFIG_BMP_16BPP=y
+      CONFIG_BMP_24BPP=y
+      CONFIG_BMP_32BPP=y
+      CONFIG_SPLASH_SOURCE=n
+      ''}
+
+      # Additional configuration (if needed)
+      ${extraConfig}
+    '';
+
+    # Inject defines for things lacking actual configuration options.
+    NIX_CFLAGS_COMPILE = lib.optionals withLogo [
+      "-DCONFIG_SYS_VIDEO_LOGO_MAX_SIZE=${toString (1920*1080*4)}"
+      "-DCONFIG_VIDEO_LOGO"
+    ];
+
+    passAsFile = [ "extraConfig" ];
+
+    configurePhase = ''
+      runHook preConfigure
+      make ${defconfig}
+      cat $extraConfigPath >> .config
+      runHook postConfigure
+    '';
+
+    installPhase = ''
+      runHook preInstall
+      mkdir -p $out
+      cp .config $out
+      ${lib.optionalString (builtins.length filesToInstall > 0) ''
+      cp ${lib.concatStringsSep " " filesToInstall} $out
+      ''}
+      runHook postInstall
+    '';
+
+    # make[2]: *** No rule to make target 'lib/efi_loader/helloworld.efi', needed by '__build'.  Stop.
+    enableParallelBuilding = false;
+
+    dontStrip = true;
+
+    meta = with lib; {
+      homepage = "https://github.com/Tow-Boot/Tow-Boot";
+      description = "Your boring SBC firmware";
+      license = licenses.gpl2;
+      maintainers = with maintainers; [ samueldr ];
+    } // meta;
+
+    passthru = {
+      inherit mkOutput patchset;
+    };
+
+  } // removeAttrs args [
+    "extraConfig"
+    "makeFlags"
+    "meta"
+    "nativeBuildInputs"
+    "patches"
+  ]);
+
+  mkOutput = commands: runCommandNoCC tow-boot.name { } ''
+    (PS4=" $ "; set -x
+    mkdir -p "$out"
+    cp -rv ${tow-boot.patchset} $out/patches
+    cp -rvt $out/ ${tow-boot}/.config
+    cp -rvt $out/ ${tow-boot}/*
+    ${commands}
+    )
   '';
 
-  installPhase = ''
-    runHook preInstall
+  patchset = runCommandNoCC "patches-for-${tow-boot.name}" { } ''
+    (PS4=" $ "; set -x
     mkdir -p $out
-    cp .config $out
-    ${lib.optionalString (builtins.length filesToInstall > 0) ''
-    cp ${lib.concatStringsSep " " filesToInstall} $out
-    ''}
-    runHook postInstall
+    cd $out
+    ${lib.concatMapStringsSep "\n" (p: "cp ${p} ./${baseNameOf (toString p)}") tow-boot.patches}
+    cat <<EOF > series
+    ${lib.concatMapStringsSep "\n" (p: "${baseNameOf (toString p)}") tow-boot.patches}
```

Comment:
This will break if two patches end up named the same.

But it's not much of an issue as if two patches end up having the same exact name, it will break loudly since the `cp` will fail overwriting the file.